.circleci/config.yml

```diff
@@ -1,41 +1,168 @@
       - checkout
       - restore_cache:
-          key: dependency-cache-v3-{{ checksum "package.json" }}
+          key: dependency-cache-v0-{{ checksum "package.json" }}
       - run: npm install
-      # Setup an NODE_ENV to production mode before compiling.
       - run:
-          name: 'Run compile in production mode'
+          name: 'Compile in production mode'
+          environment:
+            - NODE_ENV: production
           command: |
-            export NODE_ENV=production && npm run compile
-      - run: npm run vscode:package
+            npm run compile
+      - run:
+          name: 'Packaging extensions'
+          command: npm run vscode:package
       - run:
+          name: 'Staging artifacts'
           command: |
             mkdir extensions
-            find . -name "*.vsix" -type f -exec cp {} ./extensions \;
+            find packages -name "*.vsix" -type f -exec cp {} ./extensions \;
       - store_artifacts:
           path: ./extensions
       - save_cache:
-          key: dependency-cache-v3-{{ checksum "package.json" }}
+          key: dependency-cache-v0-{{ checksum "package.json" }}
+          paths:
+            - node_modules
+
+  build-win:
+    description: 'Build for Windows.'
+    steps:
+      - run: git config --global core.autocrlf false
+      - checkout
+      - restore_cache:
+          key: dependency-cache-win1-{{ checksum "package.json" }}
+      - run:
+          name: 'Install node'
+          shell: bash.exe
+          command: |
+            nvm install $NODE_VERSION
+            nvm use $NODE_VERSION
+      - run:
+          name: 'Install dependent modules'
+          shell: bash.exe
+          command: |
+            npm config set scripts-prepend-node-path true
+            npm install
+            npm install -g sfdx-cli
+            npm install -g codecov
+            npm install -g junit-merge
+      - run:
+          name: 'Compile'
+          shell: bash.exe
+          command: npm run compile
+      - save_cache:
+          key: dependency-cache-win1-{{ checksum "package.json" }}
           paths:
             - node_modules
+      - run:
+          name: 'Run lint'
+          shell: bash.exe
+          command: |
+            npm run lint
+
+jobs:
+  build-all:
+    description: 'Build artifacts in a non-Windows environment'
+    docker:
+      - image: circleci/node:10.2.0
+    environment:
+      - SFDX_URL_LINUX: https://developer.salesforce.com/media/salesforce-cli/sfdx-linux-amd64.tar.xz
+      - SFDX_AUTOUPDATE_DISABLE: true
+      - SFDX_USE_GENERIC_UNIX_KEYCHAIN: true
+      - SFDX_DOMAIN_RETRY: 300
+    steps:
+      - build-artifacts
+
+  run-tests:
+    description: 'Run tests on Windows'
+    executor: win/default
+    environment:
+      - SFDX_AUTOUPDATE_DISABLE: true
+      - SFDX_DOMAIN_RETRY: 300
+      - NODE_VERSION: '10.2.0'
+    parameters:
+      run-unit-tests:
+        type: boolean
+        default: true
+    steps:
+      - build-win
+      - when:
```

Comment:
In which scenarios do we want to run unit tests and no integration tests ?

Reply:
Unit tests and integration tests are run in parallel jobs.

Reply:
An alternative implementation would be to use the "parallelism" attribute and  CIRCLE_NODE_INDEX to separate the unit and integration tests. I felt the parameter approach was a little clearer.

Reply:
Got it, I just wanted to understand how we would run all the tests. Let's keep it as is and see how it goes, we can make changes later if needed.